invenio_rdm_records/services/services.py

```diff
@@ -191,3 +193,45 @@ def discard_pid(self, id_, identity, pid_type, pid_client=None):
             draft,
             links_tpl=self.links_item_tpl,
         )
+
+    def _lift_embargo_from(self, rdm_record):
+        """Lifts embargo from record or draft."""
+        if not rdm_record.access.embargo.lift():
+            raise EmbargoNotLiftedError(rdm_record["id"])
+        rdm_record.access.protection.record = "public"
+        rdm_record.access.protection.files = "public"
+
+    def lift_embargo(self, _id, identity):
+        """Lifts embargo from the record and updates draft."""
+
+        def draft_access_field_was_modified(draft, record):
+            """Returns True if draft's access field was modified."""
+            return draft.get('access') == record.get('access')
```

Comment:
is there a posibility of any other access field gets modified (e.g. status) and triggers an unwanted embargo lift? should we only check the embargo field? then we enter what you were mentioning of restriction change (e.g. public to restricted or viceversa) WDYT?

Reply:
I am not sure I understand the questions. This function doesn't trigger any embargo lift, in fact it's the opposite, once the embargo lift was triggered this function is meant to check if the draft access field was not modified and lift the embargo if that's the case but if the access field was modified we won't lift the embargo from the draft, only the record's embargo will be lifted, to preserve users input.
If this doesn't answer your questions, lets quickly discuss IRL.